utils/manifest-utils/src/policy.rs

```diff
@@ -90,119 +92,83 @@ impl FromStr for Match {
-            name: name.unwrap_or_else(|| petname::petname(3, "-")),
+        let store = X509StoreBuilder::new().expect("SSL works").build();
+        Self {
+            inner: Arc::new(RwLock::new(store)),
         }
     }
 }
 
 impl Keystore {
-    pub fn load(path: impl AsRef<Path>) -> anyhow::Result<Self> {
-        let path = path.as_ref();
-        let contents = std::fs::read_to_string(path)
-            .map_err(|e| anyhow::anyhow!("cannot read the keystore file: {}", e))?;
-
-        let map = contents
-            .lines()
-            .map(|s| s.trim())
-            .filter(|s| !s.is_empty() && !s.starts_with('#'))
-            .map(parse_key_entry)
-            .collect::<Result<_, _>>()?;
-
+    /// Reads PEM certificates (or certificate stacks) from `cert_dir` and creates new `X509Store`.
+    pub fn load(cert_dir: impl AsRef<Path>) -> anyhow::Result<Self> {
+        let mut store = X509StoreBuilder::new()?;
+        let cert_dir = std::fs::read_dir(cert_dir)?;
+        for dir_entry in cert_dir {
+            let cert_file = dir_entry?;
+            let cert_file = cert_file.path();
+            let mut cert_file = File::open(cert_file)?;
+            let mut cert_buffer = Vec::new();
+            cert_file.read_to_end(&mut cert_buffer)?;
+            for cert in X509::stack_from_pem(&cert_buffer)? {
+                store.add_cert(cert)?;
+            }
+        }
+        let store = store.build();
         Ok(Keystore {
-            inner: Arc::new(RwLock::new(map)),
+            inner: Arc::new(RwLock::new(store)),
         })
     }
 
-    pub fn save(&self, path: impl AsRef<Path>) -> anyhow::Result<()> {
-        let lines: Vec<String> = {
-            let inner = self.inner.read().unwrap();
-            inner.iter().map(key_entry_to_string).collect()
-        };
-
-        let mut file = OpenOptions::new()
-            .create(true)
-            .write(true)
-            .truncate(true)
-            .open(path)?;
-        lines
-            .into_iter()
-            .try_for_each(|line| file.write_all(line.as_bytes()))?;
-
-        file.flush()?;
-        file.sync_all()?;
-
-        Ok(())
-    }
-
     pub fn replace(&self, other: Keystore) {
-        let map = {
+        let store = {
             let mut inner = other.inner.write().unwrap();
-            std::mem::take(&mut (*inner))
+            std::mem::replace(&mut (*inner), X509StoreBuilder::new().unwrap().build())
         };
         let mut inner = self.inner.write().unwrap();
-        *inner = map;
-    }
-}
-
-impl Keystore {
-    pub fn contains(&self, key: &[u8]) -> bool {
-        let inner = self.inner.read().unwrap();
-        inner.contains_key(key)
+        *inner = store;
     }
 
-    pub fn insert(
+    pub fn verify_signature(
         &self,
-        key: impl Into<Box<[u8]>>,
-        scheme: Option<String>,
-        name: Option<String>,
+        cert: String,
+        sig: String,
+        sig_alg: String,
```

Comment:
It should be an enum with supported algorithms.

Reply:
Rust `openssl` library does not support list of hash algorithms either. WIll leave it as a string for now.